src/EndSwitches.cpp

```diff
@@ -0,0 +1,202 @@
+{
+    switch (_state)
+    {
+        case EndSwitchState::SWITCH_NOT_ACTIVE:
+            {
+                if (digitalRead(_minPin) == _activeState)
+                {
+                    _state            = EndSwitchState::SWITCH_AT_MINIMUM;
+                    _posWhenTriggered = _pMount->getCurrentStepperPosition(_dir);
+                    LOG(DEBUG_MOUNT,
+                        "[ENDSWITCH]: Reached minimum position on %s axis at %l. MountState is %x, %x. State is now SWITCH_AT_LIMIT",
+                        _axis == StepperAxis::RA_STEPS ? "RA" : "DEC",
+                        _posWhenTriggered,
+                        _pMount->slewStatus(),
+                        _pMount->mountStatus());
+                }
+
+                if (digitalRead(_maxPin) == _activeState)
+                {
+                    _state            = EndSwitchState::SWITCH_AT_MAXIMUM;
+                    _posWhenTriggered = _pMount->getCurrentStepperPosition(_dir);
+                    LOG(DEBUG_MOUNT,
+                        "[ENDSWITCH]: Reached maximum position on %s axis at %l. MountState is %x, %x. State is now SWITCH_AT_LIMIT",
+                        _axis == StepperAxis::RA_STEPS ? "RA" : "DEC",
+                        _posWhenTriggered,
+                        _pMount->slewStatus(),
+                        _pMount->mountStatus());
+                }
+            }
+            break;
+
+        case EndSwitchState::WAIT_FOR_STOP_AT_MAXIMUM:
+        case EndSwitchState::WAIT_FOR_STOP_AT_MINIMUM:
+        case EndSwitchState::SWITCH_AT_MAXIMUM:
+        case EndSwitchState::SWITCH_AT_MINIMUM:
+            break;
+
+        case EndSwitchState::SWITCH_SLEWING_OFF_MINIMUM:
+            {
+                if (digitalRead(_minPin) == _inactiveState)
+                {
+                    _state = EndSwitchState::SWITCH_NOT_ACTIVE;
+                    LOG(DEBUG_MOUNT,
+                        "[ENDSWITCH]: Finished moving off %s axis minimum! State is now SWITCH_INACTIVE",
+                        _axis == StepperAxis::RA_STEPS ? "RA" : "DEC");
+                }
+            }
+            break;
+        case EndSwitchState::SWITCH_SLEWING_OFF_MAXIMUM:
+            {
+                if (digitalRead(_maxPin) == _inactiveState)
+                {
+                    _state = EndSwitchState::SWITCH_NOT_ACTIVE;
+                    LOG(DEBUG_MOUNT,
+                        "[ENDSWITCH]: Finished moving off %s axis maximum! State is now SWITCH_INACTIVE",
+                        _axis == StepperAxis::RA_STEPS ? "RA" : "DEC");
+                }
+            }
+            break;
+    }
+}
+
+///////////////////////////
+//
+// checkSwitchState
+// This runs in the Mount::loop() function where all the logic
+// about stopping, parking, etc. happens.
+//
+///////////////////////////
+void EndSwitch::checkSwitchState()
+{
+    if ((_state == EndSwitchState::SWITCH_AT_MINIMUM) || (_state == EndSwitchState::SWITCH_AT_MAXIMUM))
```

Comment:
You can flip the conditional around and make it early return. Reduces the nesting logic
```C++
if (!(_state == EndSwitchState::SWITCH_AT_MINIMUM || _state == EndSwitchState::SWITCH_AT_MAXIMUM))
{
    // Don't run any logic if we're not at the stops
    return;
}
```

Reply:
I'm not a huge fan of early returns, but ok.

Reply:
In most cases early returns reduce the number of states that you need to keep in your head, makes reading code easier: https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement
(same concept as const-qualifying everything)